### PR TITLE
Form Validation

### LINKED ISSUE #27: Form Validation
**Is your feature request related to a problem? Please describe.**
There are no good/complete validations when the user inputs in the searchbar.

**Describe the solution you'd like**
The form inputs needs to have a user input validation that checks if it meets the website's requirements. If there are errors, create an error state and render it. This should be done during the onChange event, and possibly onSubmit event. Lastly, possibly create a new file that stores this functionality to be re-used following the DRY principle.

### PR DESCRIPTION
**Pull request by BlackCubes**
## Changes
1. Created a state for validation errors in App.js.
2. Added validation errors during onChange event in App.js.
3. Added any API errors to errors state in getMovieSearchResultsApi function in App.js.
4. Added setMovieName and errors as props in Header component.
5. Created checkErrors function that checks if there are any errors from error state in Header.js.
6. Added checkErrors function in condition in onSubmit event in Header.js.
6. Rendered any errors in Header.js

## Purpose
The purpose of this feature is to create user input validations when they are searching for a movie, and render any errors on the UI.

## Approach
The approach is to add a switch statement in the onChange event to validate the user input, and if the values does not meet our requirements, then store it in an errors state variable. Then, during the onSubmit event, if there are errors, then do not run the API calls. Lastly, render any errors to the user so they could correct it.

Closes #27

### CODE REVIEW

File: movie-app/src/App.js [hunk 1]
```diff
@@ -12,34 +12,57 @@ const App = () => {
...
+    const { name, value } = e.target;
+
+    switch (name) {
+      case "searchbar":
+        if (!value.length) {
+          setErrors((err) => ({ ...err, [name]: "Required." }));
+        } else if (value.length < 3) {
+          setErrors((err) => ({
+            ...err,
+            [name]: "Needs to be at least 3 characters.",
```
Comment (Lap343) [inline movie-app/src/App.js:32]: nice, nice

**Review by connielion — APPROVED** [review]
Great job for checking inputs and displaying errors.

File: movie-app/src/components/Header.js [hunk 2]
```diff
@@ -1,24 +1,51 @@
 import React from "react";
 
 function Header(props) {
+  const checkErrors = (errorObject) => {
+    // Checks to see if there are empty objects.
```
Comment (Lap343) [inline movie-app/src/components/Header.js:5]: good comments

**Review by Lap343 — APPROVED** [review]
Great job. No complaints

**Review by krferrell — APPROVED** [review]
Great commenting made for an easy read! Good work.

File: movie-app/src/App.js [hunk 3]
```diff
@@ -12,34 +12,57 @@ const App = () => {
...
+      if (movieNameData.Response === "False") {
+        setErrors((err) => ({
+          ...err,
+          searchbar: `${movieNameData.Error}, please try a more specific search term.`,
+        }));
         throw new Error(movieNameData.Error);
       }
 
-      if (movieNameData.Response === 'True') {
+      if (movieNameData.Response === "True") {
```
Comment (kmausolf) [inline movie-app/src/App.js:57]: Do they really have it capitalized in their response?  Seems odd.

**Review by gonzalespaulb — APPROVED** [review]
LGTM

**Pull request merged**